# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.h
# Commit: f615134b7349
# Full Hash: f615134b73496943a0f2b088496dc6b2e3632a0b
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Implement VideoDecoder::Reset r=padenot
#   
#   This patch implements the reset method [1] for WebCodecs' VideoDecoder
#   interface. The implementation has not been completed yet. It will be
#   completed by the following patches.
# ==============================================================================

diff -r 4e7051264cb3 -r f615134b7349 dom/media/webcodecs/VideoDecoder.h
--- a/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 04:15:01 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 04:15:01 2023 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/Attributes.h"
 #include "mozilla/DOMEventTargetHelper.h"
 #include "mozilla/ErrorResult.h"
+#include "mozilla/Result.h"
 #include "mozilla/dom/BindingDeclarations.h"
 #include "nsCycleCollectionParticipant.h"
 #include "nsWrapperCache.h"
@@ -82,9 +83,13 @@
   // VideoDecoder can run on either main thread or worker thread.
   void AssertIsOnOwningThread() const { NS_ASSERT_OWNINGTHREAD(VideoDecoder); }
 
+  Result<Ok, nsresult> Reset(const nsresult& aResult);
+
   // Constant in practice, only set in ::Constructor.
   RefPtr<WebCodecsErrorCallback> mErrorCallback;
   RefPtr<VideoFrameOutputCallback> mOutputCallback;
+
+  CodecState mState;
 };
 
 }  // namespace mozilla::dom
